homeassistant/components/kraken/__init__.py

```diff
@@ -62,11 +62,11 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         self._hass = hass
         self._config_entry = config_entry
         self._api = pykrakenapi.KrakenAPI(krakenex.API(), retry=0, crl_sleep=0)
-        self.tradable_asset_pairs = None
-        self.coordinator = None
-        self.unsub_listeners = []
+        self.tradable_asset_pairs: dict[str, str] = {}
+        self.coordinator: DataUpdateCoordinator | None = None
+        self.unsub_listeners: list[CALLBACK_TYPE] = []
```

Comment:
A future improvement: We can replace this by using `entry.async_on_unload` where needed.

Reply:
Should I replace it here or in a future PR?

Reply:
Do what you prefer. It's a small change so it's ok to add here, but also ok to do separately.

Reply:
Removed it in this PR